src/lsm/tree.zig

```diff
@@ -727,28 +813,74 @@ pub fn TreeType(comptime TreeTable: type, comptime Storage: type, comptime tree_
             assert(tree.compaction_callback != null);
 
             // compact_done() is called after all compact_tick()'s complete.
-            // This function can be triggered asynchronously or by compact_cpu() below.
             tree.compaction_io_pending -= 1;
             if (tree.compaction_io_pending == 0) tree.compact_done();
         }
 
+        /// Called at the end of each compaction tick.
         fn compact_done(tree: *Tree) void {
             assert(tree.compaction_io_pending == 0);
             assert(tree.compaction_callback != null);
 
-            // Will be true if there's any Compactions still processing.
-            var still_compacting = false;
+            // At the end of every beat:
+            // - ensure mutable table can be flushed to immutable table.
+            defer assert(tree.table_mutable.can_commit_batch(tree.options.commit_entries_max));
```

Comment:
Ran the LSM test with `accounts_to_insert_per_op` equal to `8191` or `forest_options.accounts.tree_options_id.commit_entries_max` (using `tree_options_object`'s max would overcommit the IdTree's mutable table).

On `op=7`, this assert is hit in the IdTree with its mutable table count at `32764` while the commit_entries_max is `8191`. I assume this is due to running after `compact_finish()` or `compact_drive()` from `defer`. Moving this assertion into `compact_finish()` before the callback is invoked fixes it.

Reply:
Great catch! I'll move the assertion to `compact_finish()` as you suggested.